src/components/InputV2/BigInt.tsx

```diff
@@ -0,0 +1,96 @@
+import PropTypes from 'prop-types'
+import { useState } from 'react'
+import { TextInput, TextInputProps } from './Text'
+
+const isBigInt = value => typeof value === 'bigint'
+
+export const BigIntInput = ({
+  min,
+  max,
+  value,
+  onChange,
+  onBlur,
+  setHasError,
+  ...textProps
+}: BigIntInputProps) => {
+  const [error, setError] = useState<string>('')
+  const onChangeText = (newTextValue: string) => {
+    setError('')
+    setHasError(false)
+    try {
+      onChange(BigInt(newTextValue))
+    } catch (e) {
+      // Ignore faulty BigInts, input won't update when controlled is "true"
+    }
+  }
+  const onBlurText = (newTextValue: string) => {
+    try {
+      const bigint = BigInt(newTextValue)
+      if (isBigInt(min) && bigint < min) {
+        setError(`Has to be at least ${min.toString()}`)
+        setHasError(true)
+      }
+      if (isBigInt(max) && bigint > max) {
+        setError(`Cannot be more than ${max.toString()}`)
+        setHasError(true)
+      }
+      onBlur(bigint)
+    } catch (e) {
+      setError(`Must be a whole number`)
+      setHasError(true)
+    }
+  }
+
+  return (
+    <TextInput
+      error={error}
+      type='number'
+      value={isBigInt(value) ? value.toString() : ''}
+      onChange={onChangeText}
+      onBlur={onBlurText}
+      {...textProps}
+    />
+  )
+}
+
+/**
+ * We strip certain properties from the standard
+ * text input props because we want to override them:
+ *
+ * error: set by bigint input validation
+ * type: always "number" for bigint input
+ * value: needs to be of type "BigInt" / "PropTypes.bigint"
+ * onChange: needs to take "BigInt" type argument
+ * onBlur: needs to take "BigInt" type argument
+ */
+
+export type BigIntInputProps = {
+  min?: BigInt
+  max?: BigInt
+  value?: BigInt
+  onChange: (value: BigInt) => void
+  onBlur: (value: BigInt) => void
```

Comment:
Do we definitely want to make this type required?

Reply:
Ah I see you provided default props for this. I would still mark these as optional `?` because ts will complain on the consuming app's side, it doesn't know about the defaultProps (afaik)

Reply:
That's a good call, I'll check it out

Reply:
Hey @navFooh let's mark these as optional still in the typescript types?

Reply:
Oops missed the new `Omit`